builtin/credential/approle/path_login.go

```diff
@@ -38,6 +38,9 @@ func (b *backend) pathLoginUpdate(req *logical.Request, data *framework.FieldDat
 		return logical.ErrorResponse(fmt.Sprintf("failed to validate SecretID: %s", err)), nil
 	}
 
+	// Always include the role name, for later filtering
+	metadata["role_name"] = roleName
```

Comment:
How about doing this in path_role.go handleRoleSecretIDCommon when we are persisting the metadata? We might also want to mention in docs that if SecretID metadata is not supplied, by default we add an entry there.